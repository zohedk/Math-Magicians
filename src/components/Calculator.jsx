import { useState, React } from 'react';
import calculate from '../logic/calculate';

/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-closing-bracket-location */
const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: 0 });

  const handleClick = (e) => {
    const operation = e.target.value;
    const result = calculate(state, operation);
    setState(result);
  };

  const handleTextChange = (e) => {
    setState({
      total: e.target.value,
    });
  };

  return (
    <div className="calculator-app">
      <h2>Let&apos;s do some math!</h2>
      <div className="calculator-grid">
        <div className="output">
          <div className="operand" onChange={handleTextChange}>
            {state.next || state.total || 0}
          </div>
        </div>
        <button type="button" onClick={handleClick} value="AC">
          AC
        </button>
        <button type="button" onClick={handleClick} value="+/-">
          +/-
        </button>
        <button type="button" onClick={handleClick} value="%">
          %
        </button>
        <button
          type="button"
          className="color-btn"
          onClick={handleClick}
          value="รท">
          รท
        </button>
        <button type="button" onClick={handleClick} value="7">
          7
        </button>
        <button type="button" onClick={handleClick} value="8">
          8
        </button>
        <button type="button" onClick={handleClick} value="9">
          9
        </button>
        <button
          type="button"
          className="color-btn"
          onClick={handleClick}
          value="x">
          x
        </button>
        <button type="button" onClick={handleClick} value="4">
          4
        </button>
        <button type="button" onClick={handleClick} value="5">
          5
        </button>
        <button type="button" onClick={handleClick} value="6">
          6
        </button>
        <button
          type="button"
          className="color-btn"
          onClick={handleClick}
          value="-">
          -
        </button>
        <button type="button" onClick={handleClick} value="1">
          1
        </button>
        <button type="button" onClick={handleClick} value="2">
          2
        </button>
        <button type="button" onClick={handleClick} value="3">
          3
        </button>
        <button
          type="button"
          className="color-btn"
          onClick={handleClick}
          value="+">
          +
        </button>
        <button
          type="button"
          className="span-two-border"
          onClick={handleClick}
          value="0">
          0
        </button>
        <button type="button" onClick={handleClick} value=".">
          .
        </button>
        <button
          type="button"
          className="bottom-right-btn"
          onClick={handleClick}
          value="=">
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
