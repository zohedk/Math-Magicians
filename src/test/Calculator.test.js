import React from 'react';
import TestRenderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const tree = TestRenderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
