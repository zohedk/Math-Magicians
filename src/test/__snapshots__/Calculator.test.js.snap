// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator-app"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      <div
        className="operand"
        onChange={[Function]}
      >
        0
      </div>
    </div>
    <button
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="color-btn"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="color-btn"
      onClick={[Function]}
      type="button"
      value="x"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="color-btn"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="color-btn"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="span-two-border"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="bottom-right-btn"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
